src/Validation/Validator.php

```diff
@@ -472,6 +473,22 @@ public function add(string $field, $name, $rule = [])
             if (is_array($rule)) {
                 $rule += ['rule' => $name];
             }
+            if (!is_string($name)) {
+                $name = $rule['rule'];
+                if (is_array($name)) {
+                    $name = array_shift($name);
+                }
+
+                if ($validationSet->offsetExists($name)) {
+                    $message = 'You cannot add a rule without a unique name, already existing rule found: ' . $name;
+                    throw new InvalidArgumentException($message);
+                }
+
+                deprecationWarning(
+                    'Using numeric indexes for adding validation rules is deprecated. Use string indexes instead.'
```

Comment:
```suggestion
                    'Adding validation rules without a name key is deprecated. Update rules array to have string keys.'
```

Maybe that's easier to read outside of the code?

Reply:
I am fine either way.